migration/migration.go

```diff
@@ -0,0 +1,68 @@
+package migration
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/NethermindEth/juno/db"
+)
+
+type revision func(db.DB) error
+
+var revisions = []revision{
+	revision0000,
+}
+
+func MigrateIfNeeded(targetDB db.DB) error {
```

Comment:
Can we pass in the revisions as a parameter here? The associated test appears to be unnecessarily dependent on the `revisions` global variable.

Reply:
>The associated test appears to be unnecessarily dependent on the revisions global variable.

not really, it just makes sure that first call to `MigrateIfNeeded` increments the schema version and the second call doesnt. The actual value of what schema version ends up being is not relevant.